encoding/gzip/gzip.go

```diff
@@ -0,0 +1,56 @@
+package gzip
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"sync"
+
+	"google.golang.org/grpc/encoding"
+)
+
+// RegisterNewGIZPCompressor registers gzip compressor.
+func RegisterNewGIZPCompressor() {
```

Comment:
```go
func init() {
  encoding.RegisterCompressor(new())
}

func new() {
  return &compressor{pool: sync.Pool{New: ...}}  // etc
}
```

And `compressor` should actually implement everything directly -- not inline here.